File: plugins/cordova-plugin-google-app-conversion-tracker/src/android/GoogleAppTracking.java
package com.myhealthteams.plugins;

import org.apache.cordova.CordovaPlugin;
import org.apache.cordova.CallbackContext;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import com.google.ads.conversiontracking.*;

public class GoogleAppTracking extends CordovaPlugin {

  @Override
    public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
      String conversion_id = args.getString(0);
      String tracking_label = args.getString(1);
      String tracking_value = args.getString(2);
      Boolean repeatable = args.getBoolean(3);

      if (action.equals("track")) {
        this.track(conversion_id, tracking_label, tracking_value, repeatable, callbackContext);
        return true;
      }
      return false;
    }

  private void track(String conversion_id, String tracking_label, String tracking_value, Boolean repeatable, CallbackContext callbackContext) {

    try {
      // see: https://developers.google.com/app-conversion-tracking/docs/android-conversion-tracking
      AdWordsConversionReporter.reportWithConversionId(
          this.cordova.getActivity().getApplicationContext(),
          conversion_id,
          tracking_label,
          tracking_value,
          repeatable);
    } catch(final Exception e) {
      callbackContext.error("Error in GoogleAppConversion tracking.");
    }

  }
}

File: platforms/ios/BlockTrail Wallet/Plugins/cordova-plugin-google-app-conversion-tracker/GoogleAppTracking.m
/********* GoogleAppTracking.m Cordova Plugin Implementation *******/

#import "ACTReporter.h"
#import "GoogleAppTracking.h"
#import <Cordova/CDV.h>

@implementation GoogleAppTracking

- (void)track:(CDVInvokedUrlCommand*)command
{
    NSString *conversion_id = [command.arguments objectAtIndex:0];
    NSString *tracking_label = [command.arguments objectAtIndex:1];
    NSString *tracking_value = [command.arguments objectAtIndex:2];
    BOOL repeatable = [[command.arguments objectAtIndex:3] boolValue];

    [self.commandDelegate runInBackground:^{

      // see: https://developers.google.com/app-conversion-tracking/
      [ACTConversionReporter reportWithConversionID:conversion_id label:tracking_label value:tracking_value isRepeatable:repeatable];

      CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
      [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
    }];
}

@end

